src/System.Private.CoreLib/shared/System/Memory.cs

```diff
@@ -194,12 +191,12 @@ internal Memory(object obj, int start, int length)
         /// <summary>
         /// The number of items in the memory.
         /// </summary>
-        public int Length => _length & RemoveFlagsBitMask;
+        public int Length => _length;
 
         /// <summary>
         /// Returns true if Length is 0.
         /// </summary>
-        public bool IsEmpty => (_length & RemoveFlagsBitMask) == 0;
+        public bool IsEmpty => _length == 0;
```

Comment:
```csharp
_length == 0 ? true : false
```
As the Jit isn't so good at returning the results of a test https://github.com/dotnet/coreclr/issues/12306

Reply:
I doubt that a simple test against zero suffers from the problem tracked by #12306.

Reply:
Was same issue for `string.IsNullOrEmpty` https://github.com/dotnet/coreclr/pull/17512#issuecomment-380488429 after the `cmp` it adds
```asm
+    sete     al
+    movzx    rax, al
+    test     eax, eax
```
Then uses the result of the `test` for the `jmp` rather than the result of the `cmp`

Reply:
Ah, you are right, seems to be fine

Reply:
Yep, I'm not aware of any issue involving single condition boolean expressions. `string.IsNullOrEmpty` & co. cases have multiple conditions.

Reply:
```asm
L0004: cmp dword [rcx+0x8], 0x0
L0008: jnz L0022
```